core/src/main/java/org/mule/util/metadata/ResultFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.util.metadata;
+
+import org.mule.api.metadata.FailureType;
+import org.mule.api.metadata.Result;
+import org.mule.extension.api.metadata.MetadataResolvingException;
+import org.mule.util.ExceptionUtils;
+
+public final class ResultFactory
+{
+
+    public static <T> Result<T> success(T payload)
```

Comment:
+1. Also remember the comments I did on mule-api about this not being so generic